app/src/pages/TokenPage/TokenPage.tsx

```diff
@@ -1,17 +1,127 @@
+import { useGlobals } from '../../globalsContext';
+
+const TOKEN_TRANSFER = new TokenTransfer(86323519, '0x4de7e4cbaac06e3a4fa55b8af17bf72d23f90d9d6ccace517928bd3dbb8fbf2b', '0x7Bd29408f11D2bFC23c34f18275bBf23bB716Bc7', '0xEC1B09e43100957D7623661F43364e65175eeC08', '0xEC1B09e43100957D7623661F43364e65175eeC08', '0', 6, 8999999, 98, 98889, 89889, '0x923dec2cb340dbd22a861070bb321752abec2416f24135bf473ce66fcb9479d4', new Date());
+const owner = '0x48e41913F2099300900cfcbB139F121429D38F5d';
 
 export const TokenPage = (): React.ReactElement => {
+  const { notdClient } = useGlobals();
   const routeParams = useRouteParams();
+
+  const [collectionToken, setCollectionToken] = React.useState<CollectionToken | undefined | null>(undefined);
+  const [collection, setCollection] = React.useState<Collection | undefined | null>(undefined);
   const registryAddress = routeParams.registryAddress as string;
   const tokenId = routeParams.tokenId as string;
+  const defaultImage = '/assets/icon.png';
+
+  let imageUrl = collectionToken?.imageUrl || defaultImage;
+  if (imageUrl?.startsWith('ipfs://')) {
+    imageUrl = imageUrl.replace('ipfs://', 'https://ipfs.io/ipfs/');
+  }
+
+  const updateCollectionToken = React.useCallback(async (): Promise<void> => {
+    setCollectionToken(undefined);
+    notdClient.retrieveCollectionToken(registryAddress, tokenId).then((retrievedCollectionToken: CollectionToken): void => {
+      setCollectionToken(retrievedCollectionToken);
+    }).catch((error: unknown): void => {
+      console.error(error);
+      setCollectionToken(null);
+    });
+  }, [notdClient, registryAddress, tokenId]);
+
+  React.useEffect((): void => {
+    updateCollectionToken();
+  }, [updateCollectionToken]);
+  const updateCollection = React.useCallback(async (): Promise<void> => {
+    setCollection(undefined);
+    notdClient.retrieveCollection(registryAddress).then((retrievedCollection: Collection): void => {
+      setCollection(retrievedCollection);
+    }).catch((error: unknown): void => {
+      console.error(error);
+      setCollection(null);
+    });
+  }, [notdClient, registryAddress]);
+
+  React.useEffect((): void => {
+    updateCollection();
+  }, [updateCollection]);
 
   return (
-    <Stack direction={Direction.Vertical} isFullWidth={true} isFullHeight={true} childAlignment={Alignment.Center} contentAlignment={Alignment.Start} isScrollableVertically={true}>
-      <Text variant='header3'>{`Token Page for ${registryAddress} ${tokenId}`}</Text>
+    <Stack direction={Direction.Vertical} isFullWidth={true} isFullHeight={true} childAlignment={Alignment.Start} contentAlignment={Alignment.Start} isScrollableVertically={true}>
+      {collectionToken === undefined ? (
+        <LoadingSpinner />
+      ) : collectionToken === null ? (
+        <Text variant='error'>Collection Token failed to load</Text>
+      ) : (
+        <ContainingView>
+          <ResponsiveHidingView hiddenBelow={ScreenSize.Medium}>
```

Comment:
i think you can reduce the duplcation here slightly by doing something like this:
```
<Stack responsiveDirection={{ base: Vertical, medium: Horizontal }}>
  <Image ... />
  <ResponsiveHidingView hiddenBelow={}>
     ... token name etc ...
  </ResponsiveHidingView>
  <ResponsiveHidingView hiddenAbove= {}>
     ... token name etc ...
  </ResponsiveHidingView>
</Stack>
```